### PR TITLE
refactor: refactor propose id

### PR DESCRIPTION
**Pull request by iGxnon**
Please briefly answer these questions:

proto changes: https://github.com/xline-kv/curp-proto/pull/15 https://github.com/xline-kv/xline-proto/pull/5

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

    Refactor propose id into `ProposeId(u64, u64)`

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: curp/build.rs [hunk 1]
```diff
@@ -6,7 +6,6 @@ fn main() {
         )
         .compile(
             &[
-                "./proto/common/src/message.proto",
```
Comment (Phoenix500526) [inline curp/build.rs:9]: Why do you remove this `message.proto` file

File: curp/src/server/cmd_worker/mod.rs [hunk 2]
```diff
@@ -111,10 +111,10 @@ async fn worker_exe<
                 ce.execute(cmd, entry.index).await
             };
             let er_ok = er.is_ok();
-            cb.write().insert_er(entry.id(), er);
+            cb.write().insert_er(&entry.id(), er);
```
Comment (Phoenix500526) [inline curp/src/server/cmd_worker/mod.rs:114]: We can pass the value to `insert_er` directly.

File: curp/src/server/cmd_worker/mod.rs [hunk 3]
```diff
@@ -111,10 +111,10 @@ async fn worker_exe<
...
             if !er_ok {
-                sp.lock().remove(entry.id());
-                let _ig = ucp.lock().remove(entry.id());
+                sp.lock().remove(&entry.id());
```
Comment (Phoenix500526) [inline curp/src/server/cmd_worker/mod.rs:116]: same as above

File: curp/src/server/curp_node.rs [hunk 4]
```diff
@@ -135,7 +135,7 @@ impl<C: 'static + Command, RC: RoleChange + 'static> CurpNode<C, RC> {
         let ((leader_id, term), result) = self.curp.handle_propose(Arc::clone(&cmd));
         let resp = match result {
             Ok(true) => {
-                let er_res = CommandBoard::wait_for_er(&self.cmd_board, cmd.id()).await;
+                let er_res = CommandBoard::wait_for_er(&self.cmd_board, &cmd.id()).await;
```
Comment (Phoenix500526) [inline curp/src/server/curp_node.rs:138]: Same as above

File: curp/build.rs (hunk 1 above)
Comment (iGxnon) [reply curp/build.rs:9]: It seems that `curp-command.proto` could be merged into `message.proto`. I choose to keep `curp-command.proto`.

File: xline/src/server/barriers.rs [hunk 5]
```diff
@@ -114,13 +114,13 @@ mod test {
...
                 tokio::spawn(async move {
-                    id_barrier.wait(i.to_string()).await;
+                    id_barrier.wait(ProposeId(i, i)).await;
                 })
             })
             .collect::<Vec<_>>();
         sleep(Duration::from_millis(10)).await;
         for i in 0..5 {
-            id_barrier.trigger(&i.to_string());
+            id_barrier.trigger(&ProposeId(i, i));
```
Comment (bsbds) [inline xline/src/server/barriers.rs:123]: Please also make these two method pass by value.

Comment (iGxnon) [reply xline/src/server/barriers.rs]: Done

File: curp/src/server/cmd_worker/mod.rs (hunk 2 above)
Comment (iGxnon) [reply curp/src/server/cmd_worker/mod.rs]: done

File: curp/src/server/cmd_worker/mod.rs (hunk 3 above)
Comment (iGxnon) [reply curp/src/server/cmd_worker/mod.rs:116]: It is internal used

**Review by Phoenix500526 — APPROVED** [review]

**Review by bsbds — APPROVED** [review]

**Pull request merged**